### PR TITLE
Reactor, Reactor Netty, and Spring Framework Version Upgrades

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR upgrades `reactor-core` from `3.3.x` to `3.4.x`, `reactor-netty` from `0.9.x` to `1.0.x`, and Spring Framework from `2.3` to `2.4`.

### CODE REVIEW

**Comment by backwind1233** [comment]
we are doing spring oob release now,
please do not merge the PR before the completing release,
I will inform you when the release is done.

File: sdk/core/azure-core-amqp/pom.xml
Comment (srnagar) [inline sdk/core/azure-core-amqp/pom.xml:146]: Can this be deleted?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java
Comment (srnagar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:66]: Do we have a corresponding GitHub issue filed in the Java repo?

File: sdk/storage/azure-storage-file-share/pom.xml
Comment (srnagar) [inline sdk/storage/azure-storage-file-share/pom.xml:45]: Why did we have to explicitly add these dependencies? Storage common should bring in the unreleased version of core and netty right?

**Review by srnagar — APPROVED** [review]
Please get this reviewed by the Spring and Management teams too.

File: sdk/core/azure-core-amqp/pom.xml
Comment (alzimmermsft) [reply sdk/core/azure-core-amqp/pom.xml:146]: I want to add this back in the future to prevent Azure Core AMQP from using deprecated APIs (other than those suppressed) but the new version of Reactor deprecates Processors which will require a sizeable conversion

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java
Comment (alzimmermsft) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:66]: Good question, this was copied from another branch where @kushagraThapar commit this change

File: sdk/storage/azure-storage-file-share/pom.xml
Comment (alzimmermsft) [reply sdk/storage/azure-storage-file-share/pom.xml:45]: This prevent transitive resolution errors where the additional hop results in the dependencies resolving to an unexpected version of these libraries.

File: eng/versioning/version_client.txt
Comment (alzimmermsft) [reply eng/versioning/version_client.txt:226]: Looks to be a copy and paste goof, I'll remove this once the PR is merged as this triggers a lot of pipelines.

File: sdk/appconfiguration/tests.yml
Comment (alzimmermsft) [reply sdk/appconfiguration/tests.yml:28]: This script changed the versions of Spring dependencies and ran tests again, but before 2.4 we ran both 2.2 and 2.3 but going forward 2.4 dependencies aren't compatible with those in either 2.2 or 2.3.

@backwind1233 is there plans to keep this script around and being used for a case where commit code is using dependency `2.4.x` (where x > 0) and re-run with the base `2.4.0` release?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:66]: @alzimmermsft - we can uncomment that line, and get rid of the rest of the comments. 
```suggestion
        //  This enables fast warm up of HttpClient
        //  reactorNettyClient.httpClient.warmup().block();
```

**Comment by alzimmermsft** [comment]
@amishra-dev, @kushagraThapar, @chenrujun, can I get a review on this PR, unit tests and integration tests are passing consistently with same run times as before the update.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM for mgmt.

**Review by backwind1233 — APPROVED** [review]
LGTM

**Review by mssfang — APPROVED** [review]
appconfig LGTM!

**Review by g2vinay — APPROVED** [review]

**Comment by alzimmermsft** [comment]
Performance validation for this upgrade:

**Cosmos**

Rate: calls/second

_Read Latency_

Throughput:

| Branch | Mean Rate | 1 Minute Rate | 5 Minute Rate | 15 Minute Rate |
| ------- | ------------ | --------------- | ---------------- | ----------------- |
| Current | 7327.20 | 7239.88 | 6621.34 | 6369.42 |
| Upgrade | 7414.60 | 7275.57 | 6500.45 | 6190.30 |

Latency:

Latency: ms

| Branch | Mean Rate | 1 Minute Rate | 5 Minute Rate | 15 Minute Rate | Min | Max | Mean | StdDev | Median | 75th Percentile | 95th Percentile | 98th Percentile | 99th Percentile | 99.9th Percentile |
| - | - | - | - | - | - | - | - | - | - | - | - | - | - | - |
| Current | 7327.70 | 7241.74 | 6631.46 | 6382.93 | 0.90 | 145.75 | 1.34 | 1.50 | 1.25 | 1.34 | 1.59 | 2.13 | 3.64 | 9.83 |
| Upgrade | 7415.04 | 7277.46 | 6509.82 | 6202.66 | 0.88 | 87.56 | 1.32 | 0.90 | 1.23 | 1.32 | 1.58 | 2.36 | 4.03 | 10.42 |

_Read Throughput_

| Branch | Mean Rate | 1 Minute Rate | 5 Minute Rate | 15 Minute Rate |
| ------- | ------------ | --------------- | ---------------- | ----------------- |
| Current | 7337.04 | 7243.23 | 6594.44 | 6330.77 |
| Upgrade | 7372.59 | 7268.86 | 6536.13 | 6238.55 |

_Write Throughput_

| Branch | Mean Rate | 1 Minute Rate | 5 Minute Rate | 15 Minute Rate |
| ------- | ------------ | --------------- | ---------------- | ----------------- |
| Current | 1807.79 | 1684.08 | 1533.40 | 1497.23 |
| Upgrade | 1803.42 | 1709.50 | 1591.86 | 1563.92 |

**Storage**

_Download_

| Branch | 10KB Blob | 1MB Blob | 10MB Blob |
| ------- | ----------- | ----------- | ------------ |
| Current | 1296.96 | 95.94 | 12.32 |
| Upgrade | 1321.14 | 100.86 | 12.14 |

_Upload_

| Branch | 10KB Blob | 1MB Blob | 10MB Blob |
| ------- | ----------- | ----------- | ------------ |
| Current |  |  |  |
| Upgrade |  |  |  |

**Review by gapra-msft — APPROVED** [review]

**Comment by alzimmermsft** [comment]
One live test is seeing failures consistent with other runs, not related to changes made in this PR

**Pull request merged**
